### PR TITLE
T-12181 feat: add update and create for lb and application

### PR DESCRIPTION
**Pull request by pablocampogo**
Adds update and create for loadbalancer and application.

Missing new fields of application, will be added on other PR

### CODE REVIEW

File: postgres-driver/application.go [hunk 1]
```diff
@@ -24,6 +25,32 @@ const (
...
+	VALUES (:application_id, :public_key, :address, :private_key, :version)`
+	insertPublicPocketAccountScript = `
+	INSERT into public_pocket_account (application_id, public_key, address)
+	VALUES (:application_id, :public_key, :address)`
+	insertGatewaySettingsScript = `
+	INSERT into gateway_settings (application_id, secret_key, secret_key_required, whitelist_contracts, whitelist_methods, whitelist_origins, whitelist_user_agents)
+	VALUES (:application_id, :secret_key, :secret_key_required, :whitelist_contracts, :whitelist_methods, :whitelist_origins, :whitelist_user_agents)`
+	updateApplication = `
+	UPDATE applications
+	SET name = COALESCE($1, name), updated_at = $2
```
Comment (commoddity) [inline postgres-driver/application.go]: We also need to be able to update the Status field.

For context, this is part of the RemoveApplication functionality, which updates Status and UpdatedAt, so not sure if you want to make that a separate query or if they can all be one?

File: postgres-driver/application.go [hunk 2]
```diff
@@ -160,6 +151,211 @@ func (a *dbApplication) toApplication() *repository.Application {
...
+		ApplicationID: app.ID,
+		UserID:        newSQLNullString(app.UserID),
+		Name:          newSQLNullString(app.Name),
+		ContactEmail:  newSQLNullString(app.ContactEmail),
+		Description:   newSQLNullString(app.Description),
+		Owner:         newSQLNullString(app.Owner),
+		URL:           newSQLNullString(app.URL),
+		CreatedAt:     app.CreatedAt,
+		UpdatedAt:     app.UpdatedAt,
+	}
```
Comment (commoddity) [inline postgres-driver/application.go:189]: There's a number of Application fields that the PUB sends for creating an Application that are missing here. Can you confirm if you should add them or if they are not needed?

- Dummy (bool)
- Free Tier (bool)
- FreeTierApplicationAccount (repository.FreeTierApplicationAccount)
- Status (string)

File: postgres-driver/application.go [hunk 3]
```diff
@@ -160,6 +151,211 @@ func (a *dbApplication) toApplication() *repository.Application {
...
+	return &insertFreeTierAppAccount{
+		ApplicationID: app.ID,
+		PublicKey:     newSQLNullString(app.FreeTierApplicationAccount.PublicKey),
+		Address:       newSQLNullString(app.FreeTierApplicationAccount.Address),
+		PrivateKey:    newSQLNullString(app.FreeTierApplicationAccount.PrivateKey),
+		Version:       newSQLNullString(app.FreeTierApplicationAccount.Version),
+	}
+}
+
+type insertPublicPocketAccount struct {
```
Comment (commoddity) [inline postgres-driver/application.go:258]: I don't have `PublicPocketAccount` as a field for the creation of an Application, should I add it to the create application logic in PUB? 

Assuming it just uses the address and pub key created by Pocket Go's random account creation?

File: postgres-driver/load_balancer.go [hunk 4]
```diff
@@ -73,3 +126,59 @@ func (d *PostgresDriver) ReadLoadBalancers() ([]*repository.LoadBalancer, error)
...
+}
+
+// UpdateLoadBalancerOptions struct holding possible field to update
+type UpdateLoadBalancerOptions struct {
+	Name   string
+	UserID string
+}
+
+// UpdateLoadBalancer updates fields available in options in db
+func (d *PostgresDriver) UpdateLoadBalancer(id string, options *UpdateLoadBalancerOptions) error {
```
Comment (commoddity) [inline postgres-driver/load_balancer.go]: We also need to be able to update the `UserID` and `UpdatedAt` fields, as part of the remove load balancer functionality.

Not sure if that should be a separate func?

**Review by commoddity — COMMENTED** [review]
Just a couple questions but looks good to me.

File: postgres-driver/load_balancer.go (hunk 4 above)
Comment (pablocampogo) [reply postgres-driver/load_balancer.go]: `UpdatedAt` is automatically updated with `time.Now` func, also added `UserID` field to application, it was already available in load balancer.

File: postgres-driver/application.go (hunk 2 above)
Comment (pablocampogo) [reply postgres-driver/application.go:189]: I want to add those fields in other PR, that also correctly adds them to select statement.

File: postgres-driver/load_balancer.go (hunk 4 above)
Comment (commoddity) [reply postgres-driver/load_balancer.go]: Great.

File: postgres-driver/application.go (hunk 3 above)
Comment (pablocampogo) [reply postgres-driver/application.go:258]: This field is repeated with `freeTierApplicationAccount` for now u can just don't send it, then we can discuss if we should delete it and how.

**Review by commoddity — APPROVED** [review]
Cool, looks good to me, thanks man.

**Pull request merged**